content/04-postgresql/13-reading-and-querying-data/04-optimizing-postgresql.mdx

```diff
@@ -0,0 +1,663 @@
+					             # statements running at least this number
+					             # of milliseconds;
+					             # sample fraction is determined by log_statement_sample_rate
+
+#log_statement_sample_rate = 1.0 # fraction of logged statements exceeding
+                                 # log_min_duration_sample to be logged;
+                                 # 1.0 logs all such statements, 0.0 never logs
+
+#log_transaction_sample_rate = 0.0 # fraction of transactions whose statements
+                                   # are logged regardless of their duration; 1.0 logs all
+                                   # statements from all transactions, 0.0 never logs
+. . .
+```
+
+Currently, the `log_min_duration_statement` setting is commented out with its current value set to `-1` to represent the default value.  The other settings are well-commented within the file and allow you to sample statements that are over the minimum instead of logging all of the statements.  The last setting show allows you to do sampling of statements within transactions as well.
+
+You can turn on long query logging by uncommenting the `log_min_duration_statement` and setting it to another value.  For instance, we can set it to 5 seconds to log any statements that take longer than that to complete:
+
+```
+log_min_duration_statement = 5s
+```
+
+After saving the file, you can reload your PostgreSQL server from within PostgreSQL by typing:
+
+```
+SELECT pg_reload_conf();
+```
+
+You can verify that the server is using your new settings by checking the current value again:
+
+```sql
+SELECT * FROM pg_settings WHERE name = 'log_min_duration_statement'\gx
+```
+```
+-[ RECORD 1 ]---+---------------------------------------------------------------------------
+name            | log_min_duration_statement
+setting         | 5000
+unit            | ms
+category        | Reporting and Logging / When to Log
+short_desc      | Sets the minimum execution time above which all statements will be logged.
+extra_desc      | Zero prints all queries. -1 turns this feature off.
+context         | superuser
+vartype         | integer
+source          | configuration file
+min_val         | -1
+max_val         | 2147483647
+enumvals        |
+boot_val        | -1
+reset_val       | 5000
+sourcefile      | /etc/postgresql/14/main/postgresql.conf
+sourceline      | 506
+pending_restart | f
+```
+
+Now, the `setting` field is set to 5000 and the `unit` field is set to `ms`, indicating that our setting of 5 seconds has been translated to 5000 milliseconds and applied.  The `sourcefile` line also confirms that this value is being read from the configuration file we modified.
+
+#### Logging slow queries per database
+
+Another option when trying to detect slow queries is to limit slow query logging to a specific database.  While you can  `log_min_duration_statement` can be set globally as we showed in the last section, it can also be configured at the database level.
```

Comment:
```suggestion
Another option when trying to detect slow queries is to limit slow query logging to a specific database.  While  `log_min_duration_statement` can be set globally, as we showed in the last section, it can also be configured at the database level.
```